packages/@cdktf/hcl2cdk/test/__snapshots__/iteration.test.ts.snap

```diff
@@ -27,40 +27,54 @@ class MyConvertedCode extends cdktf.TerraformStack {
+    /*In most cases loops should be handled in the programming language context and 
+    not inside of the Terraform context. If you are looping over something external, e.g. a variable or a file input
+    you should consider using a for loop. If you are looping over something only known to Terraform, e.g. a result of a data source
+    you need to keep this like it is.*/
+    const awsRoute53RecordExampleForEachIterator =
+      cdktf.TerraformIterator.fromList(
+        cdktf.Token.asAny(\`\\\\\${{
+          for dvo in \${awsAcmCertificateExample.domainValidationOptions} : dvo.domain_name => {
+            name   = dvo.resource_record_name
+            record = dvo.resource_record_value
+            type   = dvo.resource_record_type
+          }
+        }}\`)
+      );
     const awsRoute53RecordExample = new aws.route53Record.Route53Record(
       this,
       \\"example_3\\",
       {
         allowOverwrite: true,
-        name: \\"\${each.value.name}\\",
-        records: [\\"\${each.value.record}\\"],
+        name: cdktf.Token.asString(
+          cdktf.propertyAccess(awsRoute53RecordExampleForEachIterator.value, [
+            \\"name\\",
+          ])
+        ),
+        records: [
+          cdktf.Token.asString(
+            cdktf.propertyAccess(awsRoute53RecordExampleForEachIterator.value, [
+              \\"record\\",
+            ])
+          ),
+        ],
         ttl: 60,
-        type: \\"\${each.value.type}\\",
+        type: cdktf.Token.asString(
+          cdktf.propertyAccess(awsRoute53RecordExampleForEachIterator.value, [
+            \\"type\\",
+          ])
+        ),
         zoneId: cdktf.Token.asString(dataAwsRoute53ZoneExample.zoneId),
       }
     );
     /*This allows the Terraform resource name to match the original name. You can remove the call if you don't need them to match.*/
     awsRoute53RecordExample.overrideLogicalId(\\"example\\");
-    /*In most cases loops should be handled in the programming language context and 
-    not inside of the Terraform context. If you are looping over something external, e.g. a variable or a file input
-    you should consider using a for loop. If you are looping over something only known to Terraform, e.g. a result of a data source
-    you need to keep this like it is.*/
-    awsRoute53RecordExample.addOverride(
-      \\"for_each\\",
-      \`\\\\\${{
-          for dvo in \${awsAcmCertificateExample.domainValidationOptions} : dvo.domain_name => {
-            name   = dvo.resource_record_name
-            record = dvo.resource_record_value
-            type   = dvo.resource_record_type
-          }
-        }}\`
-    );
     const awsAcmCertificateValidationExample =
       new aws.acmCertificateValidation.AcmCertificateValidation(
         this,
         \\"example_4\\",
         {
-          certificateArn: awsAcmCertificateExample.arn,
+          certificateArn: cdktf.Token.asString(awsAcmCertificateExample.arn),
```

Comment:
What caused this to appear? 🤔 It should not need to exist, does it?

Reply:
I think I somehow caused a new data race condition. We need to do some extra work beforehand to get this sorted out in general. Not sure if it's a priority for 0.16 though